src/Custom/Batch/Internal/Pagination/BatchesPageEnumerator.cs

```diff
@@ -0,0 +1,109 @@
+
+#nullable enable
+
+namespace OpenAI.Batch;
+
+internal partial class BatchesPageEnumerator : PageResultEnumerator
+{
+    private readonly ClientPipeline _pipeline;
+    private readonly Uri _endpoint;
+
+    private readonly int? _limit;
+    private readonly RequestOptions _options;
+
+    private string _after;
+
+    public BatchesPageEnumerator(
+        ClientPipeline pipeline,
+        Uri endpoint,
+        string after, int? limit,
+        RequestOptions options)
+    {
+        _pipeline = pipeline;
+        _endpoint = endpoint;
+
+        _after = after;
+        _limit = limit;
+        _options = options;
+    }
+
+    public override async Task<ClientResult> GetFirstAsync()
+        => await GetBatchesAsync(_after, _limit, _options).ConfigureAwait(false);
+
+    public override ClientResult GetFirst()
+        => GetBatches(_after, _limit, _options);
+
+    public override async Task<ClientResult> GetNextAsync(ClientResult result)
+    {
+        PipelineResponse response = result.GetRawResponse();
+
+        using JsonDocument doc = JsonDocument.Parse(response.Content);
+        _after = doc.RootElement.GetProperty("last_id"u8).GetString()!;
+
+        return await GetBatchesAsync(_after, _limit, _options).ConfigureAwait(false);
+    }
+
+    public override ClientResult GetNext(ClientResult result)
+    {
+        PipelineResponse response = result.GetRawResponse();
+
+        using JsonDocument doc = JsonDocument.Parse(response.Content);
+        _after = doc.RootElement.GetProperty("last_id"u8).GetString()!;
+
+        return GetBatches(_after, _limit, _options);
+    }
+
+    public override bool HasNext(ClientResult result)
+    {
+        PipelineResponse response = result.GetRawResponse();
+
+        using JsonDocument doc = JsonDocument.Parse(response.Content);
+        bool hasMore = doc.RootElement.GetProperty("has_more"u8).GetBoolean();
+
+        return hasMore;
+    }
+
+    internal virtual async Task<ClientResult> GetBatchesAsync(string after, int? limit, RequestOptions options)
+    {
+        using PipelineMessage message = CreateGetBatchesRequest(after, limit, options);
+        return ClientResult.FromResponse(await _pipeline.ProcessMessageAsync(message, options).ConfigureAwait(false));
+    }
+
+    internal virtual ClientResult GetBatches(string after, int? limit, RequestOptions options)
+    {
+        using PipelineMessage message = CreateGetBatchesRequest(after, limit, options);
+        return ClientResult.FromResponse(_pipeline.ProcessMessage(message, options));
+    }
+
+    internal PipelineMessage CreateGetBatchesRequest(string after, int? limit, RequestOptions options)
+    {
+        var message = _pipeline.CreateMessage();
+        message.ResponseClassifier = PipelineMessageClassifier200;
+        var request = message.Request;
+        request.Method = "GET";
+        var uri = new ClientUriBuilder();
+        uri.Reset(_endpoint);
+        uri.AppendPath("/batches", false);
```

Comment:
A change just went in last week that removed the "v1" version path parameter from the default endpoint. Which means that this parameter must be added to the URI in each operation instead. Here, for example, instead "/batches", it should now be "/v1/batches". Same in the fine-tuning URIs.